File: src/main/java/it/giaquinto/springberry/model/log/LogMessage.java
package it.giaquinto.springberry.model.log;

import java.util.Date;
import java.util.Objects;

public final class LogMessage {

    private final String message;

    private final Date date;

    private final LogOut logOut;

    public LogMessage(String message) {
        this.message = message;
        this.date = new Date();
        this.logOut = LogOut.DIRECT;
    }

    public LogMessage(String message, LogOut logOut) {
        this.message = message;
        this.logOut = logOut;
        this.date = new Date();
    }

    public String getMessage() {
        return message;
    }

    public Date getDate() {
        return date;
    }

    public LogOut getLogOut() {
        return logOut;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof LogMessage that)) return false;

        if (!message.equals(that.message)) return false;
        return logOut == that.logOut;
    }

    @Override
    public int hashCode() {
        int result = message.hashCode();
        result = 31 * result + date.hashCode();
        result = 31 * result + logOut.hashCode();
        return result;
    }

    @Override
    public String toString() {
        return "LogMessage{" +
                "message='" + message + '\'' +
                ", date=" + date +
                ", logOut=" + logOut +
                '}';
    }
}

File: src/main/java/it/giaquinto/springberry/model/log/LogMessage.kt
package it.giaquinto.springberry.model.log

import java.util.*

class LogMessage {
    val message: String
    val date: Date
    val logOut: LogOut

    constructor(message: String) {
        this.message = message
        date = Date()
        logOut = LogOut.DIRECT
    }

    constructor(message: String, logOut: LogOut) {
        this.message = message
        this.logOut = logOut
        date = Date()
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o !is LogMessage) return false
        return if (message != o.message) false else logOut == o.logOut
    }

    override fun hashCode(): Int {
        var result = message.hashCode()
        result = 31 * result + date.hashCode()
        result = 31 * result + logOut.hashCode()
        return result
    }

    override fun toString(): String {
        return "LogMessage{" +
                "message='" + message + '\'' +
                ", date=" + date +
                ", logOut=" + logOut +
                '}'
    }
}